semgrep/tests/performance/test_ci_perf.py

```diff
@@ -1,203 +1,49 @@
-import pipes
 import subprocess
-from pathlib import Path
-from typing import NamedTuple
+import time
 
 import pytest
 
-from ..conftest import TESTS_PATH
 
-EXTENSIONS = {
-    "python": "py",
-    "go": "go",
-    "javascript": "js",
-}
-
-
-def sloc(path, extension=None):
-    """Quick hack in lieu of installing sloccount"""
-    if extension:
-        extension_pat = f"-name '*.{extension}'"
-    else:
-        extension_pat = ""
-    t = pipes.Template()
-    t.prepend(f"find {path} {extension_pat}", ".-")
-    t.append("xargs cat", "--")
-    t.append("wc -l", "--")
-    f = t.open("pipefile", "r")
-    return int(f.read())
-
-
-def test_sloc():
-    assert sloc(path=".", extension="py") > 0
-
-
-class RepoCase(NamedTuple):
-    url: str
-    sha: str
-    language: str
-    # used to decouple test runs when we change what the
-    # test is actually doing for benchmarking
-    version: str = "1"
-
-    @property
-    def test_id(self):
-        repo_name = self.url.split("/")[-1]
-        return f"{repo_name}-v{self.version}"
-
-
-@pytest.mark.parametrize(
-    "repo_case",
-    [
-        # SHAs of May 6, 2020
-        RepoCase(
-            "https://github.com/coinbase/node-process-lock", "dd687bc", "javascript"
-        ),
-        RepoCase("https://github.com/getsentry/sentry", "f25ea5dc", "python"),
-        RepoCase(
-            "https://github.com/bkimminich/juice-shop.git", "e03629b", "javascript"
-        ),
-    ],
-    ids=lambda case: case.test_id,
-)
-def test_default_packs(run_semgrep_in_tmp, clone_github_repo, benchmark, repo_case):
+@pytest.mark.qa
+def test_perf(clone_github_repo):
```

Comment:
Instead of testing 3 repositories in one function manually can we instead [parametrize](https://docs.pytest.org/en/latest/parametrize.html) the function to take a repo, SHA hash, and expected runtime? This will improve extensibility and allow us to easily test one set of parameters from the command line (via `pytest`'s `-k` flag). I think `xdist` will also allow us to parallelize the 3 repositories.

Reply:
Will make this change

Reply:
Actually this only tests two repositories with a shared rule repo and the repositories are executed differently (cause of --exclude in one of them) and have different expected timeout. So not a lot of shared code. Thoughts on parametrizing if/when we add more cases in the future?

Reply:
Sounds good :+1: